tests/unit/helpers/surcharges.test.js

```diff
@@ -84,127 +84,320 @@ describe('delete', () => {
+    const start = '2022-06-01T10:00:00';
+    const end = '2022-06-01T17:00:00';
+    getCustomSurcharge.onCall(1).returns(
+      [{ percentage: 25, startHour: '2022-06-01T12:00:00', endHour: '2022-06-01T14:00:00', name: 'Personnalisée' }]
+    );
+    getCustomSurcharge.onCall(0).returns([]);
+    const surcharge = {
+      evening: 10,
+      eveningStartTime: 20,
+      eveningEndTime: 23,
+      custom: 25,
+      customStartTime: 12,
+      customEndTime: 14,
+    };
+
+    const rep = SurchargesHelper.getHourlySurchargeList(start, end, surcharge);
+
+    expect(rep).toStrictEqual(
+      [{ percentage: 25, startHour: '2022-06-01T12:00:00', endHour: '2022-06-01T14:00:00', name: 'Personnalisée' }]
+    );
   });
 
-  it('should return the matching surcharge, even if surcharge values are all 0', () => {
-    const event = {
-      startDate: '2019-05-01T16:00:00.000',
-      endDate: '2019-05-01T23:00:00.000',
+  it('should return surcharge info if has evening surcharge and custom surcharge', () => {
+    const start = '2022-06-01T10:00:00';
+    const end = '2022-06-01T22:00:00';
+    getCustomSurcharge.onCall(0).returns(
+      [{ percentage: 10, startHour: '2022-06-01T20:00:00', endHour: '2022-06-01T22:00:00', name: 'Soirée' }]
+    );
+    getCustomSurcharge.onCall(1).returns(
+      [{ percentage: 25, startHour: '2022-06-01T12:00:00', endHour: '2022-06-01T14:00:00', name: 'Personnalisée' }]
+    );
+    const surcharge = {
+      evening: 10,
+      eveningStartTime: 20,
+      eveningEndTime: 23,
+      custom: 25,
+      customStartTime: 12,
+      customEndTime: 14,
     };
 
-    getCustomSurchargeStub.returnsArg(4);
-    getCustomSurchargeStub.returns();
+    const rep = SurchargesHelper.getHourlySurchargeList(start, end, surcharge);
 
-    expect(SurchargesHelper.getEventSurcharges(event, emptySurcharge)).toEqual([{
-      percentage: 0,
-      name: '1er Mai',
-    }]);
+    expect(rep).toStrictEqual([
+      { percentage: 10, startHour: '2022-06-01T20:00:00', endHour: '2022-06-01T22:00:00', name: 'Soirée' },
+      { percentage: 25, startHour: '2022-06-01T12:00:00', endHour: '2022-06-01T14:00:00', name: 'Personnalisée' },
+    ]);
   });
+});
 
-  it('should return no surcharges if none match', () => {
-    const event = {
-      startDate: '2019-05-03T16:00:00.000',
-      endDate: '2019-05-03T23:00:00.000',
+describe('getDailySurcharge', () => {
+  const surcharge = {
+    name: 'Default',
+    saturday: 12,
+    sunday: 25,
+    publicHoliday: 100,
+    twentyFifthOfDecember: 30,
+    firstOfMay: 50,
+    firstOfJanuary: 90,
+  };
+
+  it('should return null if no daily surcharge', () => {
+    const start = moment('2022-06-17T12:00:00');
+    const end = moment('2022-06-17T14:00:00');
+
+    const rep = SurchargesHelper.getDailySurcharge(start, end, surcharge);
+
+    expect(rep).toBe(null);
+  });
+
+  it('should return holiday surcharge if percentage is higher', () => {
```

Comment:
Que penses-tu de préciser de quoi le pourcentage est plus grand? 
"should return holiday surcharge if percentage is higher than hourly percentage"

Reply:
Le 1er janvier 2022 est un samedi donc ce test verifie que : 
- Si on a un samedi + un jour ferie, on prend bien la plus grande majoration 
- Que pour le 1er janvier on prend bien le pourcentage 1er janvier et pas jour ferie

On peut peut etre le renommer en "should return specific holiday surcharge and not holiday surcharge (even if surcharge is higher) nor week-end surcharge (because specific holiday surcharge > week-end surcharge) " qu'en penses-tu ? 

Sinon on peut le decouper en deux tests ^^

Reply:
J'ai mis: `should return specific holiday surcharge (not public holiday surcharge) if percentage is higher than weekend
    surcharge` , qu'en penses-tu?